addons/mass_mailing/static/src/js/snippets.editor.js

```diff
@@ -29,10 +28,23 @@ export class MassMailingSnippetsMenu extends snippetsEditor.SnippetsMenu {
      */
     setup() {
         super.setup();
-        useBus(this.props.bus, "RELOAD_SNIPPETS", ({ detail }) => {
-            this.themeParams = detail.themeParams;
-            this._loadSnippetsTemplates().then(detail.onSuccess);
-        });
+        this.mailingState = useState(this.env.mailingState);
+
+        let firstRender = true;
+        useEffect(
+            (selectedTheme) => {
+                // Avoid running this on first render as the template is already
+                // loaded for the selected theme by the normal flow.
+                if (firstRender) {
+                    firstRender = false;
+                    return;
+                }
+                this._loadSnippetsTemplates().then(() => {
```

Comment:
_loadSnippetsTemplate executes itself in the mutex, so I thought it would most likely be okay. I cannot await it in the useEffect (useEffect does not take async methods).

The only async methods that can be used in the lifecycle of an OWL Component are `onWillStart` and `onWillUpdateProps` but here, since the value is in the env, I did not use `onWillUpdateProps`. I guess that could be good reason enough to not use the env, and instead pass the variable through the props, even if wysiwyg does not use it. I just did not think there was a big risk since it's done through the mutex.

Reply:
If it's in the mutex, should be ok I guess, let's forget about this now :wink: